invenio_app_rdm/theme/views.py

```diff
@@ -105,23 +105,22 @@ def deposits_edit(pid_value):
         # Dereference relations (languages, licenses, etc.)
         draft._record.relations.dereference()
         serializer = UIJSONSerializer()
-        _record = serializer.serialize_object_to_dict(draft._record)
+        record = serializer.serialize_object_to_dict(draft.to_dict())
```

Comment:
Just a friendly note: we should start being more careful about object interaction and focus on using our public interfaces.

Reply:
Nice catch! How to serialize records is a bit messy. Actually the use of jinja and views are breaking the full flow since records do not pass by the service serializers... (Which is one of the things we wanted to fix with the *-resources modules). This was raised on the last release on two of the issues I worked on (deposit search and deposit page nor rendering due to lack of some attributes, UI serializer ones, the line you change was the fix... sorry I didnt notice about the links).

This issue, IIRC, will be tackled in this release? Is part of what Lars called "backend mess"

Reply:
Yes, I think it might be what Lars was talking about, but he was probably focusing on vocabularies for now. It's not clear to me either how to converge the serialization in API response and serialization in Jinja rendering response yet...